base/bitarray.jl

```diff
@@ -1272,29 +1272,45 @@ end
 (>>)(B::BitVector, i::Int32) = B >>> i
 (>>)(B::BitVector, i::Integer) = B >>> i
 
+function rol!(dest::BitVector, src::BitVector, i::Integer)
+    if length(dest) == length(src)
+        n = length(dest)
+        i %= n
+        i == 0 && return dest
```

Comment:
Here you need to reinstate the check `i < 0 && return ror(B, -i)` which was performed in the previous code.

You also need to change the `i==0` case like:
```
i==0 && return (src === dest ? src : copy(src))
```

otherwise `dest` will not be written in case ` dest !== src`